web/src/routes/(user)/people/[personId]/+page.svelte

```diff
@@ -0,0 +1,136 @@
+<script lang="ts">
+	import { goto } from '$app/navigation';
+	import ImageThumbnail from '$lib/components/assets/thumbnail/image-thumbnail.svelte';
+	import EditNameInput from '$lib/components/faces-page/edit-name-input.svelte';
+	import CreateSharedLink from '$lib/components/photos-page/actions/create-shared-link.svelte';
+	import DeleteAssets from '$lib/components/photos-page/actions/delete-assets.svelte';
+	import DownloadFiles from '$lib/components/photos-page/actions/download-files.svelte';
+	import MoveToArchive from '$lib/components/photos-page/actions/move-to-archive.svelte';
+	import AssetSelectContextMenu from '$lib/components/photos-page/asset-select-context-menu.svelte';
+	import AssetSelectControlBar from '$lib/components/photos-page/asset-select-control-bar.svelte';
+	import OptionAddToAlbum from '$lib/components/photos-page/menu-options/option-add-to-album.svelte';
+	import OptionAddToFavorites from '$lib/components/photos-page/menu-options/option-add-to-favorites.svelte';
+	import ControlAppBar from '$lib/components/shared-components/control-app-bar.svelte';
+	import GalleryViewer from '$lib/components/shared-components/gallery-viewer/gallery-viewer.svelte';
+	import { AppRoute } from '$lib/constants';
+	import { clickOutside } from '$lib/utils/click-outside';
+	import { handleError } from '$lib/utils/handle-error';
+	import { AssetResponseDto, api } from '@api';
+	import ArrowLeft from 'svelte-material-icons/ArrowLeft.svelte';
+	import ImageOffOutline from 'svelte-material-icons/ImageOffOutline.svelte';
+	import Plus from 'svelte-material-icons/Plus.svelte';
+	import type { PageData } from './$types';
+
+	export let data: PageData;
+
+	let isEditName = false;
+
+	let multiSelectAsset: Set<AssetResponseDto> = new Set();
+	$: isMultiSelectionMode = multiSelectAsset.size > 0;
+
+	const noop = () => {
+		// noop
+	};
+
+	const handleNameChange = async (name: string) => {
+		try {
+			isEditName = false;
+			data.person.name = name;
+			await api.personApi.updatePerson(data.person.id, { name });
+		} catch (error) {
+			handleError(error, 'Unable to save name');
+		}
+	};
+
+	const handleUnselectAsset = (assetId: string) => {
+		const asset = data.assets.find((asset: AssetResponseDto) => asset.id === assetId);
+		if (asset && multiSelectAsset.has(asset)) {
+			multiSelectAsset.delete(asset);
+		}
+	};
+</script>
+
+{#if isMultiSelectionMode}
+	<AssetSelectControlBar
+		assets={multiSelectAsset}
+		clearSelect={() => (multiSelectAsset = new Set())}
+	>
+		<CreateSharedLink />
+		<MoveToArchive onAssetArchive={(asset) => handleUnselectAsset(asset.id)} />
+		<DownloadFiles filename={data.person.name} />
+		<AssetSelectContextMenu icon={Plus} title="Add">
+			<OptionAddToFavorites />
+			<OptionAddToAlbum />
+			<OptionAddToAlbum shared />
+		</AssetSelectContextMenu>
+		<DeleteAssets onAssetDelete={handleUnselectAsset} />
```

Comment:
Deselecting assets is already handled by `clearSelect` and is automatically called when an action is triggered. The `onAssetArchive` prop should be removed, as the default behavior is correct.

The `DeleteAssets` action completely removes the asset and that may need some clarification or changing. Users could expect the asset to be only removed from the person, just like the delete button behaves in albums.

Reply:
Ah, yes that makes more sense. I missed that the action did the clear.

I think delete can work as-is for now. There is a confirmation that assets will be deleted from the library when they use it, so i think it is clear what the behavior is.